dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hdfs/HdfsContentSummaryTask.java

```diff
@@ -0,0 +1,126 @@
+import com.google.edwmigration.dumper.application.dumper.task.Task;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import com.google.edwmigration.dumper.plugin.ext.jdk.concurrent.ExecutorManager;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.ContentSummary;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HdfsContentSummaryTask implements Task<Void> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HdfsContentSummaryTask.class);
+
+  private final String clusterHost;
+  private final int port;
+
+  private enum CsvHeader {
+    Path,
+    TotalSubtreeFileSize,
+    TotalSubtreeNumberOfFiles,
+  }
+
+  HdfsContentSummaryTask(@Nonnull String clusterHost, int port) {
+    this.clusterHost = clusterHost;
+    this.port = port;
+  }
+
+  @Override
+  public String toString() {
+    return format(
+        "Write content summary of the top level directories of a hdfs %s", getTargetPath());
+  }
+
+  @Nonnull
+  @Override
+  public String getTargetPath() {
+    return "hdfs-content-summary.csv";
+  }
+
+  @CheckForNull
+  @Override
+  public Void run(TaskRunContext context) throws Exception {
+    try (OutputHandle outputHandle = context.createOutputHandle(getTargetPath());
+        Writer out = outputHandle.asTemporaryByteSink().asCharSink(UTF_8).openBufferedStream()) {
+      doRun(out, context.getExecutorService());
+    }
+    return null;
+  }
+
+  private void doRun(Writer output, Executor executorService)
+      throws IOException, ExecutionException, InterruptedException {
+    LOG.info("clusterHost: {}", clusterHost);
+    LOG.info("port: {}", port);
+
+    Configuration conf = new Configuration();
+    conf.set("fs.defaultFS", "hdfs://" + clusterHost + ":" + port + "/");
+    FileSystem fs = FileSystem.get(conf);
+
+    try (final DistributedFileSystem dfs = (DistributedFileSystem) fs;
+        final ExecutorManager execManager = new ExecutorManager(executorService);
+        final CSVPrinter csvPrinter =
+            AbstractTask.FORMAT.withHeader(CsvHeader.class).print(output)) {
+
+      String hdfsPath = "/";
+      FileStatus rootDir = fs.getFileStatus(new Path(hdfsPath));
+      FileStatus[] topLevelFiles = fs.listStatus(rootDir.getPath());
+      for (FileStatus file : topLevelFiles) {
+        // Process file or dir (in this case - just collect statistics)
+        if (file.isDirectory()) {
+          execManager.submit(
```

Comment:
I would suggest running everything in a single thread - no need for `ExecutorManager`.